src/main/java/fi/aalto/cs/apluscourses/ui/module/ModulesDialog.java

```diff
@@ -30,6 +33,13 @@ public ModulesDialog(@NotNull Project project) {
     return new Action[0];
   }
 
+  @Override
+  @Nullable
+  public JComponent getPreferredFocusedComponent() {
+    return ((JComponent) ((JBViewport) ((JBScrollPane) modulesView.getBasePanel().getComponent(1)).getComponent(
+        0)).getComponent(0));
+  }
+
```

Comment:
This relies on internal implementation details, which cannot work reliably. I got an exception:

```
java.lang.ClassCastException: class com.intellij.ui.components.JBScrollBar cannot be cast to class com.intellij.ui.components.JBViewport (com.intellij.ui.components.JBScrollBar and com.intellij.ui.components.JBViewport are in unnamed module of loader com.intellij.util.lang.PathClassLoader @7bc1a03d)
	at fi.aalto.cs.apluscourses.ui.module.ModulesDialog.getPreferredFocusedComponent(ModulesDialog.java:39)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog$MyWindowListener.lambda$windowOpened$1(DialogWrapperPeerImpl.java:868)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	...
```

Reply:
My remarks:
1. Exceptions could have been avoided by casting to JComponent everywhere, since you only care about `getComponent`. There is no need to require all the intermediate components to be e.g. particular types of scroll panes.
2. It seems like this whole expression resolves to the list view. A much safer `return modulesView.moduleListView;` should do the job instead

Reply:
Oops didn't notice there was a modulesView.moduleListView